core/templates/components/smooth-height/smooth-height-animation.component.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the smooth height change animation.
+ */
+
+import { ElementRef, HostBinding, Component, Input, OnChanges } from '@angular/core';
+import { animate, style, transition, trigger } from '@angular/animations';
+
+@Component({
+  selector: 'oppia-smooth-height',
```

Comment:
Is it worth calling this something like oppia-scroll-height-animation or oppia-scroll-height-animator, so that it's clear this is an "Animation" tag?

(Note that this would set a standard precedent for future animations.) 

What do other codebases do / what is standard practice?

(I ask because typically components are "things", and "smooth height" doesn't really sound like a "thing". That's actually the main underlying issue I see here.)

Reply:
`smooth-height` component is a workaround to achieve transition to a dynamic height. Typically, animations are not implemented using components. So, not sure what is the standard practice.

Changed selector to `oppia-scroll-height-animation`.